app/controllers/campaigns_controller.rb

```diff
@@ -10,36 +10,46 @@ def new
   end
 
   def create
-    permitted_params = CampaignParameters.new(params).permit
     @campaign = Campaign.create permitted_params
     redirect_to :campaigns, notice: 'Campaign Created'
   end
 
   def show
     @campaign = Campaign.find params['id']
-    if @campaign.active == false
+
+    unless @campaign.active?
       raise ActionController::RoutingError.new('The campaign you requested has been deactivated.')
-    end 
-    @templates = Template.where active: true
+    end
+    @templates = ActiveQuery.new(Template)
   end
 
   def edit
-    @campaign = Campaign.find params['id']
+    find_campaign
```

Comment:
You can actually make find_campaign a before filter for all these actions and then we can cut this line from all these actions